### PR TITLE
Implement Enhancing Flexibility with Customizable Default Folder Path

### PR DESCRIPTION
**Pull request by mdaneri**
Implementation Details for #1243 

Configuration via server.psd1:
Users can define custom paths for the default folders within the server.psd1 configuration file. This approach maintains Pode's configuration-driven design and ensures ease of setup. For example:

```powershell
@{
    Server = @{
        DefaultFolders = @{
            Public = 'c:\custom\public'
            Views  = 'd:\shared\views'
            Errors = 'e:\logs\errors'
        }
    }
}
```
New PowerShell Functions:
To support this feature dynamically, we propose adding two new functions:

`Set-PodeDefaultFolder`: Allows users to programmatically set the path for a default folder during runtime.

Parameters:
`-Type:` Specifies the folder type (Views, Public, or Errors).
`-Path` : Defines the new file system path for the folder.

`Get-PodeDefaultFolder`: Enables querying the current path settings for the default folders.

### CODE REVIEW

File: docs/Tutorials/Routes/Utilities/StaticContent.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ Caching is supported on static content.
 
 ## Public Directory
 
-You can place static files within the `/public` directory, at the root of your server. If a request is made for a file, then Pode will automatically check the public directory first, and if found will return the back.
+You can place static files within the `/public` directory at the root of your server, which serves as the default location for static content. However, if you need to relocate this directory, you can do so programmatically using the `Set-PodeStaticFolder` function within your server script, or specify a different location in the `server.psd1` configuration file under the `Server.DefaultFolders` property. When a request is made for a file, Pode will automatically check this designated static directory first, and if the file is found, it will be returned to the requester.
```
Comment (Badgerati) [inline docs/Tutorials/Routes/Utilities/StaticContent.md:9]: I would give a very quick example, below this paragraph, of how to set a custom public folder via `Set-PodeStaticFolder` and via the server.psd1 as well - at a minimum the latter, just so there's a reference as to the structure of the DefaultFolders block in the server.psd1 config file somewhere in the docs.

**Review by Badgerati — CHANGES_REQUESTED** [review]

File: docs/Tutorials/Routes/Utilities/StaticContent.md (hunk 1 above)
Comment (mdaneri) [reply docs/Tutorials/Routes/Utilities/StaticContent.md:9]: sure

File: src/Private/Context.ps1 [hunk 2]
```diff
@@ -814,6 +821,18 @@ function Set-PodeServerConfiguration {
...
         $Context.Server.Request.BodySize = [long]$Configuration.Request.BodySize
     }
+
+    if ($Configuration.DefaultFolders) {
+        if ($Configuration.DefaultFolders.Public) {
+            $Context.Server.DefaultFolders.Public = $Configuration.DefaultFolders.Public
+        }
+        if ($Configuration.DefaultFolders.View) {
+            $Context.Server.DefaultFolders.View = $Configuration.DefaultFolders.View
+        }
```
Comment (Badgerati) [inline src/Private/Context.ps1:831]: Just spotted,  the 3 `.View` refs should be `.Views` :)

**Review by Badgerati — CHANGES_REQUESTED** [review]

File: src/Private/Context.ps1 (hunk 2 above)
Comment (mdaneri) [reply src/Private/Context.ps1:831]: My bad.
This is the major problem with Powershell.
Nothing tells you that what you wrote is rubbish

**Review by Badgerati — APPROVED** [review]
LGTM

**Pull request merged**